clang/lib/Sema/SemaDecl.cpp

```diff
@@ -7678,9 +7678,20 @@ NamedDecl *Sema::ActOnVariableDeclarator(
       NewVD->setTSCSpec(TSCS);
   }
 
-  // Global variables with types decorated with device_global attribute must be
-  // static if they are declared in SYCL device code.
   if (getLangOpts().SYCLIsDevice) {
+    // device_global array is not allowed.
+    if (NewVD->getType()->isArrayType()) {
+      if (const Type *ArrayElementType =
+              NewVD->getType()->getPointeeOrArrayElementType()) {
+        QualType ElementQualType = QualType{ArrayElementType, 0};
+        if (isTypeDecoratedWithDeclAttribute<SYCLDeviceGlobalAttr>(
+                ElementQualType))
```

Comment:
I think you can just do - 

```suggestion
    if (const auto *AT = dyn_cast<ArrayType>(NewVD->getType())
       if (isTypeDecoratedWithDeclAttribute<SYCLDeviceGlobalAttr>(AT->getElementType()))
```

Reply:
This looks much simpler. Thank you!